llvm/test/tools/sycl-post-link/assert-property-2.ll

```diff
@@ -328,14 +319,23 @@ entry:
   ret void
 }
 
-; CHECK: _ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE7Kernel8
+; CHECK-DAG: _ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE7Kernel8
 ; Function Attrs: convergent norecurse mustprogress
 define weak_odr dso_local spir_kernel void @"_ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE7Kernel8"() local_unnamed_addr #0 {
   call spir_func void @_Z1Gv()
   call spir_func void @_Z1Hv()
   ret void
 }
 
+; CHECK-NOT: _ZTSZZ4mainENK3$_0clERN2cl4sycl7handlerEE7Kernel6
```

Comment:
Please note that `CHECK-NOT` is [bounded](https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-not-directive) by surrounding `CHECK` directives. If you want to ensure that some patters is not present in the file at all, then you should use [`--implicit-check-not`](https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-filecheck-implicit-check-not)

Reply:
Unfortunately, today I learnt that `implicit-check-not` option doesn't work with `CHECK-DAG`. At least, in a way which I got from documentation.
For example, consider the following inputs:
file:
```
a
b
c
```
ll:
```
; CHECK-DAG: a
; CHECK-DAG: c
```

If you run `FileCheck ll -input-file=./file --implicit-check-not b` then you get exit code 0 while I would expect to get non-zero.

Reply:
> Unfortunately, today I learnt that implicit-check-not option doesn't work with CHECK-DAG. At least, in a way which I got from documentation.

Yeah, I see it to now.

> I did that kernel' names print in a sorted order so that it would be easier to check them with simple CHECK:. You might have an approach better than mine.

We shouldn't be doing extra actions like sorting strings just to make our unit-tests pass. Better solution would be to still work on tests: let's just have two separate FileCheck runs on the same input, but with different prefixes ([`--check-prefix`](https://llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-filecheck-check-prefix)). The first one will check that expected names are present in the result through `CHECK-DAG`, the second one will only check that unexpected names are not present in the result.

Reply:
Thank you! Done that way.